source/material_model/rheology/peierls_creep.cc

```diff
@@ -0,0 +1,231 @@
+                               std::pow(fitting_parameters[c], glide_parameters_p[c]);
+
+              const double stress_term = 0.5 * (fitting_parameters[c] * peierls_stresses[c]) /
+                                         std::pow((prefactors[c] * std::pow(fitting_parameters[c] * peierls_stresses[c],stress_exponents[c])),( 1. / (s + stress_exponents[c])));
+
+              const double arrhenius_term = std::exp( ((activation_energies[c] + pressure * activation_volumes[c]) / (constants::gas_constant * temperature)) *
+                                                      (std::pow((1. - std::pow(fitting_parameters[c],glide_parameters_p[c])),glide_parameters_q[c])) /
+                                                      (s + stress_exponents[c]) );
+
+              const double strain_rate_term = std::pow(strain_rate, (1. / (s + stress_exponents[c])) - 1.);
+
+              viscosity = stress_term * arrhenius_term * strain_rate_term;
+              break;
+            }
+            default:
+            {
+              AssertThrow(false, ExcNotImplemented());
+              break;
+            }
+          }
+
+        return viscosity;
+      }
+
+
+
+      template <int dim>
+      void
+      PeierlsCreep<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.declare_entry ("Peierls creep flow law", "viscosity approximation",
+                           Patterns::Selection("viscosity approximation"),
+                           "Select what type of Peierls creep flow law to use. Currently, the "
+                           "only available option is an approximation to Peierls creep, which "
+                           "uses the strain rate invariant, rather than stress, as an input. ");
+        prm.declare_entry ("Prefactors for Peierls creep", "1.4e-19",
+                           Patterns::List(Patterns::Double(0)),
+                           "List of viscosity prefactors, $A$, for background material and compositional fields, "
+                           "for a total of N+1 values, where N is the number of compositional fields. "
+                           "If only one value is given, then all use the same value. "
+                           "Units: \\si{\\pascal}$^{-n_{\\text{peierls}}}$ \\si{\\per\\second}");
+        prm.declare_entry ("Stress exponents for Peierls creep", "2.0",
+                           Patterns::Anything(),
+                           "List of stress exponents, $n_{\\text{peierls}}$, for background material and compositional "
+                           "fields, for a total of N+1 values, where N is the number of compositional fields. "
+                           "If only one value is given, then all use the same value.  Units: None.");
+        prm.declare_entry ("Activation energies for Peierls creep", "320e3",
+                           Patterns::List(Patterns::Double(0)),
```

Comment:
```suggestion
                           Patterns::List(Patterns::Double(0.)),
```

Reply:
I think this comment has not been addressed yet.